dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/filter/WebAppFilterManager.java

```diff
@@ -1573,4 +1642,52 @@ private boolean findAndSetServletInfo(IFilterMapping fmInfo) {
         return false;
     }
     //issue#9386
+    
+    /**
+     * Since Servlet 6.0:
+     *  Process original URI. It rejects any path has encoded character of:
+     *  %23 (#)
+     *  %2e (.)
+     *  %2f (/)
+     *  %5c (\)
+     *  
+     *  This verification is deferred until WC can determine the request is indeed for a servlet/JSP/filter
+     */
+    private void verifyEncodedCharacter(String uri) throws IOException {
+        final boolean isTraceOn = com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled();
+        final String METHOD_NAME ="verifyEncodedCharacter";
+
+        if (isTraceOn && logger.isLoggable(Level.FINE))
+            logger.entering(CLASS_NAME, METHOD_NAME + " [" + uri + "]");
+
+        String path = uri.toLowerCase();
+        String message = null;
+
+        try {
+            if (path.contains("#") || path.contains("%23")) {
+                message = "URI has a fragment [#] character";
+            }
+            else if (path.contains("%2e")){
+                message = "URI has encoded dot [%2E] character";
+            }
+            else if (path.contains("%2f")) {
+                message = "URI has encoded forwardslash [%2F] character";
+            }
+            else if (path.contains("\\") || path.contains("%5c")){
+                message = "URI has backslash character; encoded [%5C] or not";
+            }
+
+            if (message != null) {
+                if (isTraceOn && logger.isLoggable(Level.FINE))
+                    logger.logp(Level.FINE, CLASS_NAME, METHOD_NAME, "Bad Request : " + message);
+
+                throw new IOException(nls.getFormattedMessage("bad.request.uri:.{0}", new Object[] { ((uri.length() > 128) ? (uri.substring(0, 127)) : uri) }, 
+                                "Bad request URI") + " . " + message);
```

Comment:
Shouldn't the `message` part of this be translated as well?

Reply:
add messages for translation